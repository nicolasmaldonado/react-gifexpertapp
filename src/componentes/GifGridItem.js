

export const GifGridItem = ( gif ) => {

  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={ gif.url } alt={ gif.title }/>
      <p>{ gif.title }</p>
    </div>
  )
}