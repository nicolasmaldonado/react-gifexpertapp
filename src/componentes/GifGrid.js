import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

  const { data: gifs, loading } = useFetchGifs( category );

  return (
    <>
      <h3>{ category }</h3>

      { loading && <p className='animate__flash'>Loading</p> }

      <div className='card-grid'>

        {
          gifs.map( (gif) => {
            return ( 
              <GifGridItem 
                key={ gif.id }
                { ...gif }
              />
            )
          })
        }

      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}