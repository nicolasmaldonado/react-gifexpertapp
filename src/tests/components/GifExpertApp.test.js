import { shallow } from "enzyme";
import { GifExpertApp } from "../../componentes/GifExpertApp";
import { GifGrid } from "../../componentes/GifGrid";


describe('Pruebas sobre <GifExpertApp />', () => {

  test('Debe mostrarse correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect( wrapper ).toMatchSnapshot();
  });

  test('Debe mostra una lista de X categorias', () => {
    const categories = ['Hunter x Hunter', 'Pokemon'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find(GifGrid).length ).toBe( 2 );
  })
  
  
})
