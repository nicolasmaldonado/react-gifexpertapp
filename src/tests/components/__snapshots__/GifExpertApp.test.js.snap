// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifExpertApp /> Debe mostra una lista de X categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hunter x Hunter"
      key="Hunter x Hunter"
    />
    <GifGrid
      category="Pokemon"
      key="Pokemon"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas sobre <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
